pkg/kubectl/cmd/cmd.go

```diff
@@ -257,7 +261,56 @@ var (
 )
 
 func NewDefaultKubectlCommand() *cobra.Command {
-	return NewKubectlCommand(os.Stdin, os.Stdout, os.Stderr)
+	cmd := NewKubectlCommand(os.Stdin, os.Stdout, os.Stderr)
+
+	if len(os.Args) > 1 {
+		cmdPathPieces := os.Args[1:]
+
+		// only look for suitable extension executables if
+		// the specified command does not already exist
+		if _, _, err := cmd.Find(cmdPathPieces); err != nil {
+			handleEndpointExtensions(cmdPathPieces)
+		}
+	}
+
+	return cmd
+}
+
+func handleEndpointExtensions(cmdArgs []string) {
```

Comment:
Return `error` in this method and have that `os.Exit(1)` call in `NewDefaultKubectlCommand`, otherwise you're hiding this.